### PR TITLE
Update Webauthn module to support hex/byte inputs + viem compatible types

### PR DESCRIPTION
**Pull request by KaiCode2**
# Implements
- Allows hex or bytes inputs to be passed as inputs to Webauthn module. Users no longer required to parse keys themselves
- Update installation input params to match wevm's [`P256Credential`](https://github.com/wevm/webauthn-p256/blob/main/src/types.ts#L3-L7) ([ox lib form too](https://oxlib.sh/api/WebAuthnP256))
- Update usage input params to match wevm's [`WebauthnData`](https://github.com/wevm/webauthn-p256/blob/main/src/types.ts#L15-L21) and [P256 signature](https://github.com/wevm/webauthn-p256/blob/main/src/types.ts#L23-L26) format

# Limitations
- Public key decoding only supports uncompressed key format - with or without `0x04` prefix

### CODE REVIEW

File: src/module/webauthn-validator/usage.ts [hunk 1]
```diff
@@ -1,20 +1,29 @@
 import { encodeAbiParameters, Hex } from 'viem'
+import { WebAuthnData, WebauthnSignature } from './types'
+import { parseSignature } from './utils'
+
+export type WebauthnValidatorSignature = {
```
Comment (kopy-kat) [inline src/module/webauthn-validator/usage.ts:5]: nit: could we move this into types

File: src/module/webauthn-validator/types.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { Address, type Hex } from 'viem'
+
+export type WebauthnCredential = {
```
Comment (kopy-kat) [inline src/module/webauthn-validator/types.ts:3]: could you export these types to the top level (ie through index in webauthn-validator etc)

File: src/module/webauthn-validator/utils.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { hexToBytes, bytesToHex, type Hex } from 'viem'
+import { PublicKey, WebauthnSignature } from './types'
+
+export function parsePublicKey(publicKey: Hex | Uint8Array): PublicKey {
```
Comment (kopy-kat) [inline src/module/webauthn-validator/utils.ts:4]: do you think end users will ever need to use these? if yeah then lets export them all the way too but my hunch is they wont

**Review by kopy-kat — APPROVED** [review]

File: src/module/webauthn-validator/utils.ts (hunk 3 above)
Comment (KaiCode2) [reply src/module/webauthn-validator/utils.ts:4]: Don't think they. In the event they do, they can lean on [`parseKey`](https://github.com/wevm/webauthn-p256?tab=readme-ov-file#parsepublickey) / [`parseSignature`](https://github.com/wevm/webauthn-p256?tab=readme-ov-file#parsepublickey) from the `webauthn-P256` package. But this is deliberately designed to abstract this step parsing step into an implementation detail 🤷‍♂️

**Pull request merged**